threatmodels/notation-theatmodel.md

```diff
@@ -0,0 +1,84 @@
+
+## Notation sign artifacts using remote key
+
+### Data flow
+
+A signer uses Notation to sign artifacts with the private key stored in a KMS, and store signatures in the file system or in a registry.
+
+The following diagram illustrates Notation signing artifacts stored in registries with keys stored in KMS:
+
+![Notation signs artifacts stored in registries](../media/notation-sign-remote.svg)
+
+The following diagram illustrates Notation signing artifacts as OCI image layout with keys stored in KMS:
+
+![Notation signs artifacts as OCI image layout in the filesystem](../media/notation-sign-local.svg)
+
+According to [Notary signature specification](../specs/signature-specification.md), the signing payload is a JSON document which requires the [OCI descriptor](https://github.com/opencontainers/image-spec/blob/v1.0.0/descriptor.md) of the artifact. For artifacts stored in a registry, the descriptor info can be retrieved by inspecting the [image manifest](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/manifest.md) of the artifact. For artifacts stored as an [OCI Image layout](https://github.com/opencontainers/image-spec/blob/v1.0.0/image-layout.md), the descriptor info can be retrieved from the `index.json` file from OCI Image layout.
+
+### Threats and Mitigation
+
+|   Title                      | STRIDE threat type     | Threat Status | Priority | Entity          | Description   | Mitigation   |
+| ---------------------------- | ---------------------- | ------------- | -------- | --------------- | ------------- | ------------ |
+| Compromised signing key      | Information disclosure | Mitigated     | High     | Signer          | Signing keys are compromised, and this may lead to arbitrary artifacts being signed by attackers | Rotate the signing key, and revoke the associated certificates |
+| Inaccessible Registry        | Denial of Service      | Mitigated     | High     | Registry        | Registry is not able to service incoming requests or perform up to spec, thus users are unable to publish any artifacts | Users sign artifacts stored in local file system and publish via other medias |
+| Serving tampered artifacts   | Tampering              | Mitigated     | High     | Registry        | Data stored in registries are tampered, this may lead to malicious artifacts being signed | Users sign artifacts stored in local file system, and store signatures locally or in the registry |
+| Disclosure of signature info | Information disclosure | Mitigated     | High     | Registry        | Attackers eavesdrop on registries, this may lead to the disclosure of sensitive signature info | Users should interact with trusted registries via TLS or users can sign artifacts stored in local file system and store signatures locally |
```

Comment:
@byronchien 

> Specifically for this threat/mitigation, rather than "Users should interact with trusted registries via TLS", should the mitigation be framed as "Notation interacts with registries via TLS by default"?

This is better, I will change it and try to update other mitigations

>For other threats (like "Compromised Signing Key"), is there nothing that Notation itself does to prevent this scenario? The mitigation makes it seem like there's nothing the user can do except rotate and re-sign artifacts without conveying how Notation is secure.

For this specific threat, there is no other way except rotating and re-sign artifacts since key is compromised.

Reply:
for threats that hinge on the underlying filesystem, can we call out that the user is responsible for making sure that the filesystem is secured? there might be nothing that the user can do once it happens, but there they can take action to help prevent the threat from happening.

Reply:
@byronchien Please check the latest update.